### ISSUE TITLE
Feature Request: Summarised output option

### ISSUE THREAD

**Issue by jatkinson1000** [PlasmaFAIR/fortitude#100]
Thank you for this project.

Would it be possible to have an option for condensed feedback, e.g. `file:lineno:col <error code> summary` as was the case in an older version of fortitude?

Sometimes when running over a codebase for the first time, there are many errors, often repeated and easy to correct throughout a file. The condensed display was useful here rather than the full verbose printout with source code and pointers to the offending location.

I'm thinking something like the opposite of the common `-v, --verbose` flag.
Perhaps a `-c, --concise`:
```
fortitude check -c myfile.F90
```

Would be happy to try and contribute if you can point me where to start.

**Comment by ZedThree** [comment]
Yep! Our plan is to ~~steal~~ adapt [ruff's output formats](https://docs.astral.sh/ruff/settings/#output-format), so that would look like:

```
fortitude check --output-format=concise
```

Doing it this way will let us also output annotations for GitHub's CI, JSON for an LSP server, etc.

We need one or two more pieces of infrastructure in place before we can start to use their code, but I think we're almost there!
